services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +43,54 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  } else if (thermalMgrQueueHandle == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  }
 
-  return ERR_CODE_SUCCESS;
+  if(xQueueSend(thermalMgrQueueHandle, (void *)event, 0) == pdPASS) {
+    return ERR_CODE_SUCCESS;
+  } else {
+    return ERR_CODE_QUEUE_FULL;
+  }
 }
 
+// triggered on an interrupt sent
+// interrupt means either we've exceeded the overtemp threshold or gone back down below it (hysteresis)
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t interruptEvent;
+  interruptEvent.type = THERMAL_MGR_EVENT_HANDLE_INTERRUPT;
+
+  thermalMgrSendEvent(&interruptEvent); // Register read is taken care of as temperature is checked
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  thermal_mgr_event_t queueMsg;
+  
   while (1) {
-    
+    if(xQueueReceive(thermalMgrQueueHandle, &queueMsg, 0) == pdPASS) {
+      float temp;
+      
+      error_code_t result = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+      result = ERR_CODE_INVALID_STATE;
+      
+      if (result != ERR_CODE_SUCCESS) {
+        LOG_ERROR_CODE(result);
```

Comment:
Add a continue statement after this. The rest of the code in the loop shouldn't get executed